openedx/adg/common/lib/mandrill_client/email_data.py

```diff
@@ -0,0 +1,25 @@
+"""
+Email data contains all data related to email.
+"""
+
+from django.conf import settings
+
+
+class EmailData(object):
+    """
+    Email data class that contains all data related to email
+    """
+
+    def __init__(self, template_name, useer_email, context, subject=None, attachments=None):
+        self.template_name = template_name
+        global_merge_vars = [{'name': key, 'content': context[key]} for key in context]
+
+        self.message = {
+            'from_email': settings.NOTIFICATION_FROM_EMAIL,
+            'to': [{'email': useer_email}],
+            'global_merge_vars': global_merge_vars,
+            'attachments': attachments or [],
+        }
+
+        if subject:
+            self.message.update({'subject': subject})
```

Comment:
I think emails sending from our adg platform must have subject. If we allow it to be None in init parameter then we should give default and generic subject if we didn't pass it from somewhere. Or we shouldn't allow it to be None

Reply:
As we have discussed it before we are setting subject value from mandrill web portal and if we assign some default value for 'subject' here it will override our actual subject line and we don't want that. Hope that answers your concern.

Reply:
I guess in that case when we are setting subjects in mandrill templates, we have to keep it `None` or we have to come up with a way, that we can modify the mandril templates behavior, to add add template subject when subject passed from here is `DEFAULT_SUBJECT`. For now I guess we can go with this `None` option.